File: frontend/features/practice-slots/types/groups.ts
export interface Group {
  id?: string;
  name: string;
  display_name: string;
  color: string;
  is_active: boolean;
  sort_order: number;
  created_at?: string;
  updated_at?: string;
}

export interface GroupCreate {
  name: string;
  display_name: string;
  color?: string;
  is_active?: boolean;
  sort_order?: number;
}

export interface GroupUpdate {
  name?: string;
  display_name?: string;
  color?: string;
  is_active?: boolean;
  sort_order?: number;
}

export interface ApiResponse<T> {
  success: boolean;
  data?: T;
  message?: string;
  error?: string;
}

File: backend/app/schemas/groups.py
from datetime import datetime
from typing import Optional
from uuid import UUID
from pydantic import BaseModel, ConfigDict


class GroupBase(BaseModel):
    name: str
    display_name: str
    color: Optional[str] = "#3B82F6"
    is_active: bool = True
    sort_order: int = 0


class GroupCreate(GroupBase):
    pass


class GroupUpdate(BaseModel):
    name: Optional[str] = None
    display_name: Optional[str] = None
    color: Optional[str] = None
    is_active: Optional[bool] = None
    sort_order: Optional[int] = None


class GroupResponse(GroupBase):
    id: UUID
    created_at: Optional[datetime] = None
    updated_at: Optional[datetime] = None
    model_config = ConfigDict(from_attributes=True)
